overlord/devicestate/devicestate.go

```diff
@@ -293,6 +293,28 @@ func CanManageRefreshes(st *state.State) bool {
 	return false
 }
 
+func getAllRequiredSnapsForModel(model *asserts.Model) map[string]bool {
+	reqSnaps := model.RequiredSnaps()
+	// +4 for (snapd, base, gadget, kernel)
+	required := make(map[string]bool, len(reqSnaps)+4)
+	for _, snap := range reqSnaps {
+		required[snap] = true
+	}
+	if model.Base() != "" {
```

Comment:
Should this just say `model.Base() != "core" {`?

Reply:
core is not a base though